ci/travis_before_script_cpp.sh

```diff
@@ -31,7 +31,8 @@ CPP_DIR=$TRAVIS_BUILD_DIR/cpp
 
 CMAKE_COMMON_FLAGS="\
 -DARROW_BUILD_BENCHMARKS=ON \
--DCMAKE_INSTALL_PREFIX=$ARROW_CPP_INSTALL"
+-DCMAKE_INSTALL_PREFIX=$ARROW_CPP_INSTALL \
+-DCMAKE_CXX_STANDARD=11"
```

Comment:
agreed. I will remove it here and add a follow up patch to fix it more generically, and enforce that we must use c++11